### PR TITLE
[WIP] Async wrapper

### PR DESCRIPTION
**Pull request by nelstrom**
Fixes issue #3.

### CODE REVIEW

**Comment by prabirshrestha** [comment]
@nelstrom `a:message` in neovim is an array where value is separated by `\n`. async.vim normalizes it to array even for vim8 so you need to join by `\n`. I do this in the async.vim sample (will update the docs to be clear about this)

```vim
function! s:job_err_cb(name, channel, message) abort
  echohl WarningMsg
  echom a:name . ': ' . join(a:message, "\n")
  echohl None
endfunction
```

**Comment by nelstrom** [comment]
@prabirshrestha thanks for the advice!

**Comment by nelstrom** [comment]
@prabirshrestha I'm a bit confused about what arguments are provided to the callbacks. In NeoVim, it looks like the callback receives arguments `s:JobHandler(job_id, data, event)` (from [`:help job-control-usage`](https://neovim.io/doc/user/job_control.html#job-control-usage)). I'm not sure what would be the equivalent in Vim8. Or does async.vim normalise Vim8 callback args to be the same as in NeoVim?

**Comment by prabirshrestha** [comment]
async.vim normalizes vim8 to neovim so you get the the same as neovim. `s:JobHandler(job_id, data, event)`.
Feel free to send a PR in async.vim to add docs or improve README.md

**Comment by k-takata** [comment]
@nelstrom 
Can you try [async-wrapper branch](https://github.com/k-takata/minpac/tree/async-wrapper)?

File: autoload/minpac/job.vim [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        endif
+    endif
+endfunction
+
+function! s:job_start(cmd, opts) abort
+    let l:jobtypes = s:job_supported_types()
+    let l:jobtype = ''
+
+    if has_key(a:opts, 'type')
+        if type(a:opts.type, v:t_string)
```
Comment (prabirshrestha) [inline autoload/minpac/job.vim]: i fixed the bug on upstream async.vim. You would want to sync this with latest master.

**Comment by k-takata** [comment]
I created a new PR based on this. See #5.

**Pull request closed without merging**